cmd/do-agent/run.go

```diff
@@ -51,7 +53,11 @@ func run(w metricWriter, l limiter, dec decorate.Decorator, g gatherer) {
 		dec.Decorate(mfs)
 		log.Debug("stats decorated in %s", time.Since(start))
 
-		err = w.Write(mfs)
+		aggregated, err := aggregate.Aggregate(mfs, aggregateSpec)
```

Comment:
Do we maybe want to decorate after aggregation to avoid decorating more than we need to?

Reply:
It could be a good idea but, it might be better to do it afterwards as is that way metrics with mismatched cases and stuff still get properly aggregated in

Reply:
or does that not matter 🤔

Reply:
I'd prefer to decorate first, so we know what metrics we're expecting. There is one decorator that makes all metrics lowercase

Reply:
yeah, we need to decorate first anyhow to make sure we aggregate appropriately